### PR TITLE
Support downloading system tests from plugins

### PR DESCRIPTION
**Pull request by tsteur**
### Description:

Works same way as syncing UI test screenshots.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/CoreConsole/Commands/DevelopmentSyncProcessedSystemTests.php [hunk 1]
```diff
@@ -34,6 +34,10 @@ protected function configure()
         $this->setDescription('For Piwik core devs. Copies processed system tests from travis artifacts to local processed directories');
         $this->addArgument('buildnumber', InputArgument::REQUIRED, 'Travis build number you want to sync, eg "14820".');
         $this->addOption('expected', 'e', InputOption::VALUE_NONE, 'If given file will be copied in expected directories instead of processed');
+        $this->addOption('repository', 'r', InputOption::VALUE_OPTIONAL, 'Repository name you want to sync screenshots for.', 'matomo-org/matomo');
```
Comment (sgiehl) [inline plugins/CoreConsole/Commands/DevelopmentSyncProcessedSystemTests.php:37]: I guess when another repository is given we actually could skip calling `updatePluginsFiles`, as it won't be available for repos other then matomo

Comment (tsteur) [reply plugins/CoreConsole/Commands/DevelopmentSyncProcessedSystemTests.php:37]: added a check for this 👍

File: plugins/CoreConsole/Commands/DevelopmentSyncProcessedSystemTests.php [hunk 2]
```diff
@@ -34,12 +34,19 @@ protected function configure()
...
+        $this->addOption('http-password', '', InputOption::VALUE_OPTIONAL, 'the HTTP AUTH password (for premium plugins where artifacts are protected)');
+
     }
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
         $this->updateCoreFiles($input, $output);
-        $this->updatePluginsFiles($input, $output);
+
+        if (!$input->getOption('repository')) {
```
Comment (sgiehl) [inline plugins/CoreConsole/Commands/DevelopmentSyncProcessedSystemTests.php]: won't that always return a value? Guess we would need to compare it's `matomo-org/matomo`

Comment (tsteur) [reply plugins/CoreConsole/Commands/DevelopmentSyncProcessedSystemTests.php]: 👍 updated it

**Comment by sgiehl** [comment]
@tsteur The download now actually works. But it will always sync the files into the core test folder. As normally someone wants to sync the files into the plugin directly, might be good to change that a bit. I'll try to add another commit to make that possible.

**Pull request merged**